### YamlMime:PythonClass
uid: azure.mgmt.cdn.operations.ValidateOperations
name: ValidateOperations
fullName: azure.mgmt.cdn.operations.ValidateOperations
module: azure.mgmt.cdn.operations
inheritances:
- builtins.object
summary: "> [!WARNING]\n> DO NOT instantiate this class directly.\n>\n> \n>\n> Instead,\
  \ you should access the following operations through\n>\n> <xref:azure.mgmt.cdn.CdnManagementClient>'s\n\
  >\n> <xref:validate> attribute.\n>"
constructor:
  syntax: ValidateOperations(*args, **kwargs)
methods:
- uid: azure.mgmt.cdn.operations.ValidateOperations.secret
  name: secret
  summary: Validate a Secret in the profile.
  signature: 'secret(validate_secret_input: _models.ValidateSecretInput, *, content_type:
    str = ''application/json'', **kwargs: Any) -> _models.ValidateSecretOutput'
  parameters:
  - name: validate_secret_input
    description: The Secret source. Is either a model type or a IO type. Required.
    isRequired: true
    types:
    - <xref:azure.mgmt.cdn.models.ValidateSecretInput>
    - <xref:azure.mgmt.cdn.operations.IO>
  - name: content_type
    description: 'Body Parameter content-type. Known values are: ''application/json''.

      Default value is None.'
    types:
    - <xref:str>
  - name: cls
    description: A custom type or function that will be passed the direct response
    types:
    - <xref:callable>
  return:
    description: ValidateSecretOutput or the result of cls(response)
    types:
    - <xref:azure.mgmt.cdn.models.ValidateSecretOutput>
  exceptions:
  - type: azure.core.exceptions.HttpResponseError
attributes:
- uid: azure.mgmt.cdn.operations.ValidateOperations.models
  name: models
  signature: models = <module 'azure.mgmt.cdn.models' from 'C:\\hostedtoolcache\\windows\\Python\\3.11.1\\x64\\Lib\\site-packages\\azure\\mgmt\\cdn\\models\\__init__.py'>
